builder/views.py

```diff
@@ -161,6 +170,27 @@ def put(self, request: Request, key: str, **kwargs) -> HttpResponse:
 
         return JsonResponse({"success": False, "error": form.errors})
 
+    @login_required_method(redirect_url=None)
+    def put(self, request: Request, remote_id: int, **kwargs) -> HttpResponse:
+        data = QueryDict(request.body, mutable=True)
+        if not data:
+            return JsonResponse({"success": False, "error": "No POST parameters given"})
+
+        course = get_object_or_404(Course, remote_id=remote_id)
+        if not course.has_access(request, Permission.WRITE):
+            return JsonResponse({"success": False, "error": f"No access to instance {course.remote_id}"})
+
+        response = self._check_access(request, False)
+        if response is not None:
+            return response
+
+        form = CourseForm(data, instance=course)
+        if form.is_valid():
+            course = form.save(request)
+            return JsonResponse(self._get(request, course))
+
+        return JsonResponse({"success": False, "error": form.errors})
+
```

Comment:
Please combine this with the above method, we don't want to duplicate code. See the `get` method comment.